app/objects/c_operation.py

```diff
@@ -328,33 +328,36 @@ async def _get_all_possible_abilities_by_agent(self, data_svc):
         return {a.paw: abilities for a in self.agents}
 
     def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors, agent_ran):
-        variables = re.findall(r'#{(.*?)}', self.decode_bytes(ability.test), flags=re.DOTALL) if ability.test else []
         if ability.ability_id in agent_ran:
             return
-        elif not agent.trusted:
+
+        ability_has_platform = any(executor for executor in ability.executors if executor.platform == agent.platform)
+        valid_executors = ability.get_executors(agent.platform, agent_executors)
+
+        any_facts_fulfilled = False
+        for executor in valid_executors:
+            facts = re.findall(r'#{(.*?)}', executor.command, flags=re.DOTALL) if executor.command else []
+            if all(fact in op_facts for fact in facts):
+                any_facts_fulfilled = True
+
+        if not agent.trusted:
             return dict(reason='Agent untrusted', reason_id=self.Reason.UNTRUSTED.value,
                         ability_id=ability.ability_id, ability_name=ability.name)
-        elif ability.platform != agent.platform:
+        elif not ability_has_platform:
             return dict(reason='Wrong platform', reason_id=self.Reason.PLATFORM.value, ability_id=ability.ability_id,
                         ability_name=ability.name)
-        elif ability.executor not in agent_executors:
+        elif not valid_executors:
             return dict(reason='Executor not available', reason_id=self.Reason.EXECUTOR.value,
                         ability_id=ability.ability_id, ability_name=ability.name)
-        elif variables and not all(op_fact in op_facts for op_fact in variables):
+        elif not any_facts_fulfilled:
             return dict(reason='Fact dependency not fulfilled', reason_id=self.Reason.FACT_DEPENDENCY.value,
                         ability_id=ability.ability_id, ability_name=ability.name)
         elif not agent.privileged_to_run(ability):
             return dict(reason='Ability privilege not fulfilled', reason_id=self.Reason.PRIVILEGE.value,
                         ability_id=ability.ability_id, ability_name=ability.name)
-        else:
-            if (ability.platform == agent.platform and ability.executor in agent_executors
-                    and ability.ability_id not in agent_ran):
-                if state != 'finished':
-                    return dict(reason='Operation not completed', reason_id=self.Reason.OP_RUNNING.value,
-                                ability_id=ability.ability_id, ability_name=ability.name)
-                else:
-                    return dict(reason='Agent untrusted', reason_id=self.Reason.UNTRUSTED.value,
-                                ability_id=ability.ability_id, ability_name=ability.name)
+        elif state != 'finished':
+            return dict(reason='Operation not completed', reason_id=self.Reason.OP_RUNNING.value,
```

Comment:
I see that we do this throughout the function, but do you know why we use the enum member `value` rather than the enum member itself?

Reply:
I do not.

Reply:
My hunch is that it's some [de]serialization or compatibility issue. But yeah, feels off. Outside the scope of this issue to address it though.